programs/ssl/ssl_client2.c

```diff
@@ -1698,22 +1717,38 @@ int main( int argc, char *argv[] )
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     if( opt.key_opaque != 0 )
     {
-        psa_algorithm_t psa_alg, psa_alg2;
+        psa_algorithm_t psa_alg, psa_alg2 = 0;
+        psa_key_usage_t usage = PSA_KEY_USAGE_SIGN_HASH;
 
-        if( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_ECKEY )
+        if( strcmp( opt.key_opaque_alg1, DFL_KEY_OPAQUE_ALG ) == 0 )
         {
-            psa_alg = PSA_ALG_ECDSA( PSA_ALG_ANY_HASH );
-            psa_alg2 = PSA_ALG_NONE;
+            if( mbedtls_pk_get_type( &pkey ) == MBEDTLS_PK_ECKEY )
+            {
+                opt.key_opaque_alg1 = "ecdsa-sign";
+                opt.key_opaque_alg2 = "none";
+            }
+            else
+            {
+                opt.key_opaque_alg1 = "rsa-sign-pkcs1";
+                opt.key_opaque_alg2 = "rsa-sign-pss";
+            }
         }
-        else
+
+        if ( strcmp( opt.key_opaque_alg1, DFL_KEY_OPAQUE_ALG ) != 0 )
```

Comment:
Sorry if I'm missing something obvious, but at this point I think we know this condition will be false: if `opt.key_opaque_alg1` was equal to `DFL_KEY_OPAQUE_ALG` on line 1725, then is has been changed to something else, and if it wasn't equal on line 1725, then it's not going to be equal now.

Reply:
Yes, that is true. As now I'm handling default parameters by setting `opt.key_opaque_algX` `key_opaque_set_alg_usage()` is always called. 
On client side `opt.key_opaque_algX` are always set and never have default values, which is not true on server side.